web_dialog_size/static/src/js/web_dialog_size.js

```diff
@@ -1,34 +1,41 @@
 odoo.define('web_dialog_size.web_dialog_size', function (require) {
 'use strict';
 
-var Model = require('web.DataModel');
+var rpc = require('web.rpc');
 var Dialog = require('web.Dialog');
 
-Dialog.include({
+var DialogSize = Dialog.include({
 
-    init: function (parent, options) {
+    willStart: function () {
         var self = this;
-        this._super.apply(this, arguments);
-        self.$modal.find('.dialog_button_extend').on('click', self.proxy('_extending'));
-        self.$modal.find('.dialog_button_restore').on('click', self.proxy('_restore'));
-
-        new Model('ir.config_parameter').query(['key', 'value']).
-        filter([['key', '=', 'web_dialog_size.default_maximize']]).all().then(function(default_maximize) {
-            if (default_maximize.length && default_maximize[0].value == 1) {
-                self._extending();
-            } else {
-                self._restore();
-            }
+        var res = this._super.apply(this, arguments).then(function () {
+            self.$modal.find('.dialog_button_extend').on('click', self.proxy('_extending'));
+            self.$modal.find('.dialog_button_restore').on('click', self.proxy('_restore'));
+
+            rpc.query({
+                model: 'web.dialog.size.backend',
+                method: 'get_default_maximize',
+            }).then(function(dialog_size_data) {
+                if (dialog_size_data.default_maximize == 1) {
+                    self._extending();
+                } else {
+                    self._restore();
+                }
+            });
         });
+        return res;
     },
 
     open: function() {
-        var res = this._super.apply(this, arguments);
-        this.$modal.draggable({
-            handle: '.modal-header',
-            helper: false
-        });
-        return res;
+        var self = this;
+        this._super.apply(this, arguments);
+        if (self.$modal != undefined) {
+            self.$modal.draggable({
+                handle: '.modal-header',
+                helper: false
+            });
+        }
+        return self;
```

Comment:
why did you stop to return the super value ?

Reply:
I removed this.